### PR TITLE
Backend changes for Vulnerabilities Timeline page

### PR DESCRIPTION
**Pull request by jonnydh**

### CODE REVIEW

File: app/uk/gov/hmrc/vulnerabilities/controllers/VulnerabilitiesTimelineController.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+class VulnerabilitiesTimelineController @Inject()(
+ cc: ControllerComponents,
+ vulnerabilitiesTimelineRepository: VulnerabilitiesTimelineRepository
+) (implicit ec: ExecutionContext) extends BackendController(cc)
+  with Logging {
+
+  implicit val fmt = VulnerabilitiesTimelineCount.apiFormat
+
+  def getTimelineCountsForService(service: Option[String], team: Option[String], vulnerability: Option[String], from: Instant, to: Instant): Action[AnyContent] = Action.async {
+    vulnerabilitiesTimelineRepository.getTimelineCountsForService(service, team, vulnerability, from, to).map {
```
Comment (beny23) [inline app/uk/gov/hmrc/vulnerabilities/controllers/VulnerabilitiesTimelineController.scala:40]: OBSERVATION: is that method name still right - I'm assuming `service` can be `None`

File: app/uk/gov/hmrc/vulnerabilities/model/TimelineEvent.scala [hunk 2]
```diff
@@ -43,3 +43,37 @@ object ServiceVulnerability {
...
   }
 }
+
+case class VulnerabilitiesTimelineCount(
+  weekBeginning       : Instant,
+  actionRequired      : Int,
+  investigationOngoing: Int,
+  noActionRequired    : Int,
+  uncurated           : Int,
+  total               : Int
```
Comment (beny23) [inline app/uk/gov/hmrc/vulnerabilities/model/TimelineEvent.scala:53]: CLARIFICATION: do we still need the counts keyed out per curation status? won't the filter deal with that?

File: app/uk/gov/hmrc/vulnerabilities/persistence/VulnerabilitiesTimelineRepository.scala [hunk 3]
```diff
@@ -34,13 +39,17 @@ class VulnerabilitiesTimelineRepository @Inject()(
 ) extends PlayMongoRepository(
   collectionName = "vulnerabilitiesTimeline",
   mongoComponent = mongoComponent,
-  domainFormat   = ServiceVulnerability.mongoFormat,
+  domainFormat   = TimelineEvent.mongoFormat,
   indexes        = Seq(
-    IndexModel(compoundIndex(descending("weekBeginning"), descending("service"), descending("id")), IndexOptions().unique(true)),
-    IndexModel(ascending("weekBeginning"), IndexOptions().expireAfter(2 * 365, TimeUnit.DAYS))
+    IndexModel(compoundIndex(descending("weekBeginning"), descending("service"), descending("id")),IndexOptions().unique(true).background(true)),
+    IndexModel(Indexes.ascending("weekBeginning"),IndexOptions().expireAfter(2 * 365, TimeUnit.DAYS).background(true)),
```
Comment (beny23) [inline app/uk/gov/hmrc/vulnerabilities/persistence/VulnerabilitiesTimelineRepository.scala:45]: CLARIFICATION: Can we make the expiration configurable?

File: app/uk/gov/hmrc/vulnerabilities/persistence/VulnerabilitiesTimelineRepository.scala [hunk 4]
```diff
@@ -54,4 +63,44 @@ class VulnerabilitiesTimelineRepository @Inject()(
     )
     collection.bulkWrite(bulkWrites).toFuture().map(_ => ())
   }
+
+  val Quoted = """^\"(.*)\"$""".r
+
+  def getTimelineCountsForService(service: Option[String], team: Option[String], vulnerability: Option[String], from: Instant, to: Instant): Future[Seq[VulnerabilitiesTimelineCount]] = {
```
Comment (beny23) [inline app/uk/gov/hmrc/vulnerabilities/persistence/VulnerabilitiesTimelineRepository.scala:69]: AMBER: Wouldn't it make sense to filter for curation status here?

**Review by beny23 — DISMISSED** [review]

File: app/uk/gov/hmrc/vulnerabilities/controllers/VulnerabilitiesTimelineController.scala (hunk 1 above)
Comment (jonnydh) [reply app/uk/gov/hmrc/vulnerabilities/controllers/VulnerabilitiesTimelineController.scala]: resolved :)

File: app/uk/gov/hmrc/vulnerabilities/model/TimelineEvent.scala (hunk 2 above)
Comment (jonnydh) [reply app/uk/gov/hmrc/vulnerabilities/model/TimelineEvent.scala]: resolved :)

File: app/uk/gov/hmrc/vulnerabilities/persistence/VulnerabilitiesTimelineRepository.scala (hunk 3 above)
Comment (jonnydh) [reply app/uk/gov/hmrc/vulnerabilities/persistence/VulnerabilitiesTimelineRepository.scala:46]: Agreed on slack that its fine to leave in code, as seems to be how we do it everywhere else

File: app/uk/gov/hmrc/vulnerabilities/persistence/VulnerabilitiesTimelineRepository.scala (hunk 4 above)
Comment (jonnydh) [reply app/uk/gov/hmrc/vulnerabilities/persistence/VulnerabilitiesTimelineRepository.scala]: resolved :) Logic is now in backend rather than frontend

**Review by Shnick — APPROVED** [review]

**Pull request merged**